lightkurve/targetpixelfile.py

```diff
@@ -856,21 +922,30 @@ def from_fits_images(images, position=None, size=(10, 10), extension=None,
                                                n_rows=size[0],
                                                n_cols=size[1],
                                                target_id=target_id)
+        # Find middle image to use as a reference
+        mid_img = images[int(len(images)/2)-1]
+
         for idx, img in tqdm(enumerate(images), total=len(images)):
-            if isinstance(img, fits.ImageHDU):
-                hdu = img
-            elif isinstance(img, fits.HDUList):
-                hdu = img[extension]
-            else:
-                hdu = fits.open(img)[extension]
+            hdu = KeplerTargetPixelFile.get_hdu(img, extension)
             if idx == 0:  # Get default keyword values from the first image
                 factory.keywords = hdu.header
             if position is None:
                 cutout = hdu
             else:
                 cutout = Cutout2D(hdu.data, position, wcs=WCS(hdu.header),
                                   size=size, mode='partial')
-            factory.add_cadence(frameno=idx, flux=cutout.data, header=hdu.header)
+
+            img_data = KeplerTargetPixelFile.get_WCS(mid_img,hdu,position,extension)
+            # Add calculated pos corr shifts to the keywords dictionary
+            hdu.header['POSCORR1'] = img_data[1]
+            hdu.header['POSCORR2'] = img_data[2]
+            # Information on the center of the TPF file to be created, used to create the second header extension
+            factory.keywords['MID_COL'] = img_data[3]
+            factory.keywords['MID_ROW'] = img_data[4]
+            factory.keywords['RA_CTR'] = img_data[5]
+            factory.keywords['DEC_CTR'] = img_data[6]
+
```

Comment:
Let's not make new header keywords if we don't have to and if people won't use them.